lib/adapters/class-searchpress.php

```diff
@@ -132,18 +132,28 @@ public function add_search_suggest_to_indexed_post_data( $data, $sp_post ) {
 
 	/**
 	 * A callback for the sp_search_query_args filter. Adds aggregations to the
-	 * request so that the response will include aggregation buckets.
+	 * request so that the response will include aggregation buckets, and
+	 * filters by any aggregations set.
 	 *
 	 * @param array $es_args The request args to be filtered.
 	 * @return array The filtered request args.
 	 */
 	public function add_aggs_to_es_query( $es_args ): array {
 		// Add our aggregations.
 		foreach ( $this->get_aggregations() as $aggregation ) {
+			// Add aggregations to the request so buckets will be returned with results.
 			$request = $aggregation->request();
 			if ( ! empty( $request ) ) {
 				$es_args['aggs'][ $aggregation->get_query_var() ] = $request;
 			}
+
+			// Add any set aggregations to the query so results are properly filtered.
+			if ( ! empty( $es_args['query']['bool']['must'] ) && is_array( $es_args['query']['bool']['must'] ) ) {
```

Comment:
this feels like a bug waiting to happen. why does `must` have to be populated for this to work?

an ideal solution would to check if `$es_args['query']['bool']` is set, and if so, add this filter to the array of `$es_args['query']['bool']['filter']` if it's set, or add `filter` if it's not set. if `$es_args['query']['bool']` isn't set, and `$es_args['query']` is set, then this could replace `$es_args['query']` and add the existing query as a `bool.must`. the DSL class could be expanded to add a method that modifies queries for you, and every agg could leverage that.

if that's too much work for this PR, I'd say just check to see if `$es_args['query']['bool']` is set before adding the filter.

Reply:
The approach I've taken here (and in the VIP Enterprise Search adapter) is to be extremely conservative when modifying DSL since there are several potential places that could modify DSL (e.g., other filters, changes to the plugin itself). If it's not in a shape that we expect, don't modify it. The failure case here is that the filter doesn't get applied, vs. mangling a different type of query and potentially breaking something else on the site.

I'll explore the filter route—that sounds reasonably safe overall. Agreed that we could build in a system in the DSL class for modifying queries (and that itself could make a good Composer package).